httptransport/indexer_v1.go

```diff
@@ -0,0 +1,344 @@
+	inner http.Handler
+	srv   indexer.Service
+}
+
+var _ http.Handler = (*IndexerV1)(nil)
+
+// ServeHTTP implements http.Handler.
+func (h *IndexerV1) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	start := time.Now()
+	wr := responserecorder.NewResponseRecorder(w)
+	defer func() {
+		if f, ok := wr.(http.Flusher); ok {
+			f.Flush()
+		}
+		zlog.Info(r.Context()).
+			Str("remote_addr", r.RemoteAddr).
+			Str("method", r.Method).
+			Str("request_uri", r.RequestURI).
+			Int("status", wr.StatusCode()).
+			Dur("duration", time.Since(start)).
+			Msg("handled HTTP request")
+	}()
+	h.inner.ServeHTTP(wr, r)
+}
+
+func (h *IndexerV1) indexReport(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case http.MethodPost:
+	case http.MethodDelete:
+	default:
+		apiError(w, http.StatusMethodNotAllowed, "method disallowed: %s", r.Method)
+		return
+	}
+	ctx := r.Context()
+	defer r.Body.Close()
+	dec := codec.GetDecoder(r.Body)
+	defer codec.PutDecoder(dec)
+	switch r.Method {
+	case http.MethodPost:
+		state, err := h.srv.State(ctx)
+		if err != nil {
+			apiError(w, http.StatusInternalServerError, "could not retrieve indexer state: %v", err)
+			return
+		}
+		var m claircore.Manifest
+		if err := dec.Decode(&m); err != nil {
+			apiError(w, http.StatusBadRequest, "failed to deserialize manifest: %v", err)
+			return
+		}
+		if m.Hash.String() == "" || len(m.Layers) == 0 {
+			apiError(w, http.StatusBadRequest, "bogus manifest")
+			return
+		}
+		next := path.Join(r.URL.Path, m.Hash.String())
+
+		w.Header().Add("link", fmt.Sprintf(linkIndex, next))
+		w.Header().Add("link", fmt.Sprintf(linkReport, path.Join(VulnerabilityReportPath, m.Hash.String())))
+		validator := `"` + state + `"`
+		if unmodified(r, validator) {
+			w.WriteHeader(http.StatusPreconditionFailed)
+			return
+		}
+
+		// TODO Do we need some sort of background context embedded in the HTTP
+		// struct?
+		report, err := h.srv.Index(ctx, &m)
+		if err != nil {
+			apiError(w, http.StatusInternalServerError, "failed to start scan: %v", err)
+			return
+		}
+
+		w.Header().Set("etag", validator)
+		w.Header().Set("location", next)
+		defer writerError(w, &err)()
+		w.WriteHeader(http.StatusCreated)
+		enc := codec.GetEncoder(w)
+		defer codec.PutEncoder(enc)
+		err = enc.Encode(report)
+	case http.MethodDelete:
+		var ds []claircore.Digest
+		if err := dec.Decode(&ds); err != nil {
+			apiError(w, http.StatusBadRequest, "failed to deserialize bulk delete: %v", err)
+			return
+		}
+		ds, err := h.srv.DeleteManifests(ctx, ds...)
+		if err != nil {
+			apiError(w, http.StatusInternalServerError, "could not delete manifests: %v", err)
+			return
+		}
+		zlog.Debug(ctx).
+			Int("count", len(ds)).
+			Msg("manifests deleted")
+		defer writerError(w, &err)()
+		w.WriteHeader(http.StatusOK)
+		enc := codec.GetEncoder(w)
+		defer codec.PutEncoder(enc)
+		err = enc.Encode(ds)
+	}
+}
+
+const (
+	linkIndex  = `<%s>; rel="https://projectquay.io/clair/v1/index_report"`
```

Comment:
No, it's the relation type: https://www.rfc-editor.org/rfc/rfc5988#section-5.3

Reply:
Oh I interpreted that as if you use the `link-extension` then it had to be an absolute URI otherwise it should describe the relationship between the current resource and the link in the header, I was looking at this: https://datatracker.ietf.org/doc/html/rfc2068#section-19.6.2.4